src/ol_infrastructure/applications/open_metadata/__main__.py

```diff
@@ -0,0 +1,209 @@
+            check_id="open-metadata-instance-db",
+            interval="10s",
+            name="open-metadata-instance-id",
+            timeout="60s",
+            status="passing",
+            tcp=open_metadata_db.db_instance.address.apply(
+                lambda address: f"{address}:{open_metadata_db_config.port}"
+            ),
+        )
+    ],
+    opts=consul_provider,
+)
+
+# Create a vault policy to allow dagster to get to the secrets it needs
+open_metadata_server_vault_policy = vault.Policy(
+    "open-metadata-server-vault-policy",
+    name="open-metadata-server",
+    policy=Path(__file__)
+    .parent.joinpath("open_metadata_server_policy.hcl")
+    .read_text(),
+)
+## Begin block for migrating to pyinfra images
+consul_datacenter = consul_stack.require_output("datacenter")
+
+# Pulumi k8s omd help stuff. Still struggling with syntax.
+
+kubeconfig = cluster_stack.require_output("kube_config")
+cluster_name = cluster_stack.require_output("cluster_name")
+VERSIONS = cluster_stack.require_output("versions")
+
+k8s_provider = kubernetes.Provider(
+    "k8s-provider",
+    kubeconfig=kubeconfig,
+)
+
+open_metadata_application = kubernetes.helm.v3.Release(
+    f"{cluster_name}-open-metadata-application-helm-release",
+    kubernetes.helm.v3.ReleaseArgs(
+        name="open-metadata-application",
+        chart="open-metadata-application",
+        version=VERSIONS["OPEN_METADATA"],
+        namespace="applications",
+        cleanup_on_fail=True,
+        repository_opts=kubernetes.helm.v3.RepositoryOptsArgs(
+            repo="https://helm.open-metadata.org",
+        ),
+        values={
+            "openmetadata": {
+                "config": {
+                    "elasticsearch": {
+                        # "auth": {
+                        # We don't export any of the auth stuff? WHY? -cap
+                        #     "username": "elastic",
+                        #     "password": {
+                        #         "secretRef": "elasticsearch-password",
+                        #         "secretKey": "password",
+                        #     },
+                        # },
+                        # Umm. Don't we need to specify the domain? -cap
+                        "host": opensearch_cluster["endpoint"],
+                        "port": opensearch_cluster["port"],
+                    },
+                    "database": {
+                        "host": open_metadata_db.db_instance.address,
+                        "port": open_metadata_db_config.port,
+                        "databaseName": open_metadata_db_config.db_name,
+                        "auth": {
+                            "username": open_metadata_db_config.username,
+                            "password": {
+                                "secretRef": open_metadata_db_config.password_secret_name,
+                                "secretKey": open_metadata_db_config.password_secret_key,
+                            },
```

Comment:
These values should either be coming from Vault, or we might take a look at using IAM auth to RDS since OpenMetadata supports that out of the box.

Reply:
I will figure out the vault dynamic secret resources and add it to this PR. It will be a good example. At the moment the only thing I've used on the EKS stacks is static secrets.